internal/definitions/handler_identity_claim.go

```diff
@@ -246,5 +246,23 @@ func (dh *definitionHandler) handleIdentityClaim(ctx context.Context, state *cor
 		event := core.NewEvent(core.EventTypeIdentityConfirmed, identity.Namespace, identity.ID, nil, core.SystemTopicDefinitions)
 		return dh.database.InsertEvent(ctx, event)
 	})
+
+	if identity.Type == core.IdentityTypeNode {
+		nodeDID, err := dh.identity.GetLocalNodeDID(ctx)
+		if err != nil {
+			return HandlerResult{Action: core.ActionRetry}, err
```

Comment:
This seems to require that we've successfully published our node identity at the point this runs, in all circumstances.

Have you done a though exercise on new network setup to confirm that is true in all cases (syncing a long lived chain and publishing your own identity after many others for example)?

Reply:
> This seems to require that we've successfully published our node identity at the point this runs, in all circumstances.

So that was not my intent, but lemme re-read the code. The _new_  `GetLocalNodeDID` func is meant to be just building the DID string from the multiparty manager's local node field, as opposed to the `GetLocalNode` func that did already exist but to your point actually cared about the local node being in the DB/caches bc it had been registered or was registering.

I had not thought that the `multiparty.LocalNode` field comes purely from config, not if the local node has been registered. From what I've seen testing locally (and the E2E passing) these assumptions seem correct.

Reply:
Yes, confirmed that multiparty manager's `LocalNode` is purely a config field:
```
func (mm *multipartyManager) LocalNode() LocalNode {
	return mm.config.Node
}
```

so this code does not require a node to be registered in order to work. Its just seeing if the local node DID we expect based on config is being claimed/updated.

Reply:
Great stuff, thanks for checking through that